sys/arm64/arm64/trap.c

```diff
@@ -299,11 +299,11 @@ print_registers(struct trapframe *frame)
 
 	for (reg = 0; reg < nitems(frame->tf_x); reg++) {
 		printf(" %sx%d: %16lx\n", (reg < 10) ? " " : "", reg,
-		    frame->tf_x[reg]);
+		    (uint64_t)frame->tf_x[reg]);
```

Comment:
I assume there is a follow-up patch to print these as capregs? If not there should be a TODO here.

Reply:
Again, we’re committing workarounds here for things that will be changed (I hope). This is needless churn that polluted the history. We do not need that merged.

Reply:
The line needs to be changed again anyway once we print caps, I don't think it matters much whether we touch it once or twice. Especially since we won't be able to compile the kernel without this change unless we turn of -Wformat.

Reply:
I can add a TODO so I can get back to it after image activator changes are ready.

Reply:
So either:
(a) turn off -Wformat as an early commit and then make this line print the capability
(b) use a version of the compiler with Alex's cherry-picked patches
It's not hard to avoid the problem here.

Reply:
I really don't like option a). This is a genuine bug. It happens to work correctly in this case since the vararg happens to be passed in registers, but if this was a printf with more arguments we would print the wrong value for later args.

Reply:
There's no bug here? You'd make the format string be a capability for CHERI kernels and the argument be a capability, and just turn off -Wformat because the compiler doesn't yet know the format string is valid?

Reply:
tf_x is a `uintcap_t` in the `__has_feature(capabilities)` case, so -Wformat complains.

Reply:
Yes, so use `%#.16lp` in that case, ie _CHERI_PRINTF_CAP_FMT.

Reply:
I missed the "then make this line print the capability" part of option a). I thought you were suggesting we simply disable warnings until printing capregs has been implemented.